File: Week-01/10-Daily Temperatures/Assem.cpp
// Author: Assem
// Problem:  Daily Temperatures
// Link: https://leetcode.com/problems/daily-temperatures/description/
// Time: O(n²) | Space: O(n)

#include <bits/stdc++.h>
using namespace std;

// Input: temperatures = [73,74,75,71,69,72,76,73]
// Output: [1,1,4,2,1,1,0,0]

class Solution
{
public:
    vector<int> dailyTemperatures(vector<int> &temperatures)
    {
        vector<int> days;

        for (int i = 0; i < temperatures.size(); i++)
        {
            int distance = 0;

            for (int j = i + 1; j < temperatures.size(); j++)
            {
                if (temperatures[j] > temperatures[i])
                {
                    distance = j - i;
                    break;
                }
            }

            days.push_back(distance);
        }

        return days;
    }
};

File: Week-01/10-Daily Temperatures/Assem.cs
// Author: Assem
// Problem:  Clear Digits
// Link: https://leetcode.com/problems/clear-digits/description
// Time: O(n) | Space: O(n)

class Solution
{
public:
    string clearDigits(string s)
    {
        string result;

        for (char c : s)
        {
            if (isdigit(c))
            {
                if (!result.empty())
                    result.pop_back();
            }
            else
            {
                result.push_back(c);
            }
        }

        return result;
    }
};
